06-building-a-program-start-to-finish/writing-tests/output/invertedIndexSpec.js

```diff
@@ -0,0 +1,183 @@
+const validFile = require('./books/validFile.json');
+const simpleValidFile = require('./books/simpleValidFile.json');
+const empty = require('./books/empty.json');
+const emptyText = require('./books/emptyText.json');
+const notArray = require('./books/notArray.json');
+const wrongData = require('./books/wrongData.json');
+
+const index = new InvertedIndex();
+
+describe('InvertedIndex', () => {
+  describe('Constructor', () => {
+    it('can create an instance of InvertedIndex', () => {
+      expect(typeof index).toEqual('object');
+    });
+    it('initializes properties correctly', () => {
+      expect(index.files).toEqual([]);
+      expect(index.showAllFiles).toBeFalsy();
+      expect(index.indexed).toEqual({});
+      expect(index.showAllFilesSearch).toBeFalsy();
+      expect(index.searchAllResult).toEqual({});
+    });
+  });
+
+  describe('ValidateFile', () => {
+    it('should return false if an invalid file was uploaded', () => {
+      expect(InvertedIndex.validateFile(empty)).toBeFalsy();
+      expect(InvertedIndex.validateFile(emptyText)).toBeFalsy();
+      expect(InvertedIndex.validateFile(notArray)).toBeFalsy();
+      expect(InvertedIndex.validateFile(wrongData)).toBeFalsy();
+    });
+    it('should not return false if a valid file was uploaded', () => {
+      expect(InvertedIndex.validateFile(validFile)).not.toBeFalsy();
+      expect(InvertedIndex.validateFile(simpleValidFile)).not.toBeFalsy();
+      expect(typeof InvertedIndex.validateFile(validFile)).toEqual('object');
+    });
+    it('should return the file content if the file is valid', () => {
+      expect(InvertedIndex.validateFile(validFile)).toEqual(validFile);
+      expect(InvertedIndex.validateFile(simpleValidFile))
+      .toEqual(simpleValidFile);
+    });
+  });
+
+  describe('Tokenize', () => {
+    it('should return an array of unique words', () => {
+      expect(InvertedIndex.tokenize('tony is humble tony'))
+      .toEqual(['tony', 'is', 'humble']);
+    });
+    it('should remove invalid characters', () => {
+      expect(InvertedIndex.tokenize('$A%^n*del)a %i=s @a$w&es$om#e%'))
+      .toEqual(['andela', 'is', 'awesome']);
+    });
+    it('should return lowercase words', () => {
+      expect(InvertedIndex.tokenize('HEllO WORLD')).toEqual(['hello', 'world']);
+    });
+  });
+
+  describe('CreateIndex', () => {
+    beforeAll(() => {
+      this.filename = 'simpleValidFile.json';
+    });
+
+    it('creates an index object from the file content', () => {
+      expect(index.createIndex(simpleValidFile, this.filename)).toBeTruthy();
+      expect(typeof index.getIndex(this.filename)).toEqual('object');
```

Comment:
I see you're using `getIndex` method here and you don't test this method till the next suite. Are you aware your test execute sequentially?

If for some reason `getIndex` doesn't work properly your tests would fail in the `createIndex` suite first which could lead to confusion.